torchani/data/new.py

```diff
@@ -0,0 +1,458 @@
+            chunks[i] = (chunks[i][0], chunks[i][1].reshape(chunks[i][1].shape[0], -1, 3))
+
+        properties = {'energies': datas[2][0].flatten()}
+
+        # return: [chunk1, chunk2, ...], {"energies", "force", ...} in which chunk1=(species, coordinates)
+        # e.g. chunk1 = [[1807, 21], [1807, 21, 3]], chunk2 = [[193, 50], [193, 50, 3]]
+        # 'energies' = [2000, 1]
+        return chunks, properties
+
+    def __len__(self):
+        return self.length
+
+    @staticmethod
+    def sort_list_with_index(inputs, index):
+
+        return [inputs[i] for i in index]
+
+    @staticmethod
+    def split_list_with_size(inputs, split_size):
+
+        output = []
+        # split_size = np.array(split_size)
+        for i, value in enumerate(split_size):
+            start_index = np.sum(split_size[:i])
+            stop_index = np.sum(split_size[:i + 1])
+            output.append(inputs[start_index:stop_index])
+        return output
+
+    def pad_and_convert_to_tensor(self, inputs, padding_value=0, no_padding=False):
+
+        if no_padding:
+            for i, input_tmp in enumerate(inputs):
+                inputs[i] = torch.from_numpy(np.stack(input_tmp)).to(self.device)
+        else:
+            for i, input_tmp in enumerate(inputs):
+                inputs[i] = torch.nn.utils.rnn.pad_sequence([torch.from_numpy(b) for b in inputs[i]],
+                                                            batch_first=True,
+                                                            padding_value=padding_value).to(self.device)
+        return inputs
+
+    def find_threshold(self, threshold_max=100):
+        batch_indices = slice(0, self.batch_size)
+        batch_indices_shuffled = self.shuffled_index[batch_indices]
+
+        batch_species = [self.data_species[i] for i in batch_indices_shuffled]
+
+        num_atoms_each_mole = [b.shape[0] for b in batch_species]
+        atoms = torch.tensor(num_atoms_each_mole, dtype=torch.int32)
+
+        output, count = torch.unique(atoms, sorted=True, return_counts=True)
+        counts = torch.cat((output.unsqueeze(-1).int(), count.unsqueeze(-1).int()), dim=-1)
+
+        print('=> choose a reasonable threshold to split chunks')
+        print('format is [chunk_size, chunk_max]')
+
+        for b in range(0, threshold_max, 1):
+            test_chunk_size_list, test_chunk_max_list = split_to_chunks(counts, chunk_threshold=b * self.batch_size * 20)
+            size_max = []
+            for i, _ in enumerate(test_chunk_size_list):
+                size_max.append([list(test_chunk_size_list[i].numpy())[0],
+                                list(test_chunk_max_list[i].numpy())[0]])
+            print('chunk_threshold = {}'.format(b))
+            print(size_max)
+
+    def release_h5(self):
+        import gc
+        del self.data_species
+        del self.data_coordinates
+        del self.data_energies
+        gc.collect()
+
+
+def ShuffledDataset(file_path,
```

Comment:
Rename it to `something_like_this`, title format is used for types. I think a good approach is to rename this to `create_dataloader` and rename `TorchData` into `ShuffledDataset`.

Reply:
`ShuffledDataset` is for user to ues, vs `CachedDataset`
`TorchData` is not visible to user

Reply:
OK, then how about `shuffled_dataloader`?

Reply:
Good, but not consistent with `CachedDataset`
TorchData means data returned is one molecule of data, which saved as tensor, not numpy

Reply:
The user only needs to use it, the use of CachedDataset and ShuffledDataset is exactly the same if he does not use index to use it.

Reply:
OK sounds good to me now. Thanks for clarifying.